Revamp README

ChrisBrownie55 opened the pull request:
Work-in-progress: I'm currently revamping the README page for this. You can take a look at the stuff I've added to this branch, it's mostly just adding more descript paragraphs and more examples.

Any suggestions, advice, or critique are welcome.

dayhaysoos commented:
Just a heads up, I had a conversation with a Stripe developer advocate and it was highly recommended that I change the name of this library to avoid potential branding issues.

I'm going to rename to `use-shopping-cart`

ChrisBrownie55 commented:
@dayhaysoos I'll change the name of it in the README here if you haven't changed it in master already. Let me know if there are any other README changes that need to be made to accommodate this #63

dayhaysoos commented:
@ChrisBrownie55 there's the documentation branch that is gonna need quite a few changes, but I can take care of it.

ChrisBrownie55 commented:
Okay so things I need to finish on the README and things I already have:

- [x] Updated wording for each example
- [x] Removed library specific DOM components (i.e. `<Image>`, `<Box>`, and `<Button>`) and replaced them with simple DOM elements
- [x] Full use of `<CartProvider>` props
- [ ] Separate examples for each part of the hook
  - [x] How to add an item
  - [x] Display the cart
    - [x] Reduce item by one
    - [x] Increase item by one
    - [ ] Set quantity (requires code refactor from #61)
    - [ ] Remove product from cart
- [ ] Show how/when to use `cartCount`
- [ ] `handleCartHover` and `handleCloseCart`, `handleCartClick`, and `shouldDisplayCart`???
- [ ] `storeLastClicked`, `lastClicked`???

dayhaysoos commented:
I think it'll be okay for now to omit `handleCarthover`, handleCloseCart`, `handleCartClick`, `shouldDisplayCart`, `storeLastClicked`, and `lastClicked`

We haven't written any examples for that quite yet.

doingandlearning commented on README.md:
```diff
@@ -1,72 +1,73 @@
+
+At the root level of your application (or the highest point you'll be using Stripe from), wrap your components in a `<CartProvider>`.
 
-Remember to add your public Stripe key, in this example it's added in in the environment `process.env.REACT_APP_STRIPE_API_PUBLIC`.
```
This seems to have been removed from the docs but I think this or something similar should be here or in the installation. Something like,


> If using the library with Stripe, the package expects a REACT_APP_STRIPE_API_PUBLIC to be avaiable in your .env file. An example .env file can be found in .env.example.

ChrisBrownie55 replied:
@doingandlearning I converted this line into a comment just above where `loadStripe` is called. I don't believe that having a longer explanation for that part helps at all.

Also, as far as I know, the package does not expect a `REACT_APP_STRIPE_API_PUBLIC` environment variable in a `.env` file as `use-shopping-cart` doesn't read or use any environment variables, however, if someone wishes to they can use `.env` in their own project for this purpose they can but they don't have to. They only need to provide their public Stripe key.

This is the line I have:
```js
// Remember to add your public Stripe key
const stripePromise = loadStripe(process.env.REACT_APP_STRIPE_API_PUBLIC)
```

Let me know if that's sufficient enough to help the developer understand that they need to remember to use their public Stripe key when calling `loadStripe`.

If they don't know how to add environment variables we could provide links to more in-depth tutorials on that matter, if that helps.

dayhaysoos replied:
I think that's. great idea, and adding env variable links sounds great, too.

I'm planning on making a lot of examples/starters and egghead videos to educate people on how to use this library.

ChrisBrownie55 commented:
Creating a symbolic link for `/use-shopping-cart/README.md` to point to `/REAMDE.md`. Hopefully, it works, otherwise, we might end up with two readmes for the same thing or a hyperlink from one to the other.

ChrisBrownie55 commented:
Seems to look fine from both `/` and `/use-shopping-cart/` directories of this repo 👍

ChrisBrownie55 commented:
@doingandlearning Let me know if my solution was good enough.
@dayhaysoos I'll merge once you guys approve 👍

ChrisBrownie55 commented:
Just exported `toCurrency` from `index.js` as it's used in a lot of examples and definitely seems to be useful. If we don't plan on doing that, let me know so I can undo that and edit the docs to reflect that.

ChrisBrownie55 commented:
Also installed `@types/stripe-v3` because our JSDoc comment for `<CartProvider>` said that `stripe` was of type `any` and now it points to the actual type definition of the `Stripe` object

ChrisBrownie55 commented:
@all-contributors please add @ChrisBrownie55 for doc
